### PR TITLE
Update CONTRIBUTION.md with pre-commit + black-primer instructions

### PR DESCRIPTION
**Pull request by cooperlees**
- Inform people how to run primer and alter it's config
- Link to main documentation

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -34,12 +34,26 @@ If you haven't used `pipenv` before but are comfortable with virtualenvs, just r
 `pip install pipenv` in the virtualenv you're already using and invoke the command above
 from the cloned _Black_ repo. It will do the correct thing.
 
-Before submitting pull requests, run tests with:
+Before submitting pull requests, run lints, tests + CI with:
```
Comment (hugovk) [inline CONTRIBUTING.md:37]: CI is run only when pushing, right? Maybe leave it out from here?

```suggestion
Before submitting pull requests, run lints and tests with:
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -34,12 +34,26 @@ If you haven't used `pipenv` before but are comfortable with virtualenvs, just r
...
-```
+```sh
+$ pre-commit run -a
 $ python -m unittest
+$ black-primer [-k -w /tmp/black_test_repos]
 ```
 
+## black-primer
+
+`black-primer` is used by CI to pull down well know _Black_ formatted projects and see
```
Comment (hugovk) [inline CONTRIBUTING.md:47]:
```suggestion
`black-primer` is used by CI to pull down well-known _Black_ formatted projects and see
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -34,12 +34,26 @@ If you haven't used `pipenv` before but are comfortable with virtualenvs, just r
...
+if we get soure code changes. It will error on formatting changes or errors. Please run
+before pusing your PR to see if you get the actions you would expect from _Black_ with
```
Comment (hugovk) [inline CONTRIBUTING.md:49]:
```suggestion
before pushing your PR to see if you get the actions you would expect from _Black_ with
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -24,7 +24,7 @@ You can use any operating system. I am using macOS myself and CentOS at work.
 
 Install all development dependencies using:
 
-```
+```sh
```
Comment (hugovk) [inline CONTRIBUTING.md:27]: Prefer ` ```console` when there's a `$` prompt

```suggestion
```console
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -34,12 +34,26 @@ If you haven't used `pipenv` before but are comfortable with virtualenvs, just r
 `pip install pipenv` in the virtualenv you're already using and invoke the command above
 from the cloned _Black_ repo. It will do the correct thing.
 
-Before submitting pull requests, run tests with:
+Before submitting pull requests, run lints, tests + CI with:
 
-```
+```sh
```
Comment (hugovk) [inline CONTRIBUTING.md:39]:
```suggestion
```console
```

File: CONTRIBUTING.md (hunk 1 above)
Comment (cooperlees) [reply CONTRIBUTING.md:37]: Yeah I agree. I was just trying to say run Unit and Integration tests which you've achieved better. Thanks!

File: CONTRIBUTING.md (hunk 2 above)
Comment (cooperlees) [reply CONTRIBUTING.md:47]: Good catch.

**Comment by cooperlees** [comment]
That batch apply is rad. Thanks @GitHub.

**Pull request merged**